### PR TITLE
Merge projects

### PR DESCRIPTION
**Pull request by rogeralsing**
Merging Mailbox and Router into core Proto.Actor project

### CODE REVIEW

File: src/Proto.Actor/Mailbox/Mailbox.cs [hunk 1]
```diff
@@ -10,7 +10,7 @@
 
 namespace Proto.Mailbox
 {
```
Comment (cpx86) [inline src/Proto.Actor/Mailbox/Mailbox.cs]: What's the reason for changing these to public?

Comment (cpx86) [reply src/Proto.Actor/Mailbox/Mailbox.cs]: Same for BoundedMailboxQueue

Comment (rogeralsing) [reply src/Proto.Actor/Mailbox/Mailbox.cs]: We currently rely on the internalvisibleto thing for testing.
We could ofc still do that for the core lib. it just seems hacky.

Should we keep internal?

Comment (alexeyzimarev) [reply src/Proto.Actor/Mailbox/Mailbox.cs]: I think internalvisibleto is ok as long as it won't create a lot of issues with signing

Comment (cpx86) [reply src/Proto.Actor/Mailbox/Mailbox.cs]: For low-level unit tests I think InternalsVisibleTo is the way to go. Better than exposing internal classes in the API surface.

Comment (cpx86) [reply src/Proto.Actor/Mailbox/Mailbox.cs]: (tbh in general I think we could and probably should make some other public classes internal)

Comment (rogeralsing) [reply src/Proto.Actor/Mailbox/Mailbox.cs]: Brought back internal and also moved mailbox and router tests into same project as the actor tests.

**Pull request merged**